utxoscanner.go

```diff
@@ -0,0 +1,415 @@
+package neutrino
+
+import (
+	"container/heap"
+	"sync"
+	"sync/atomic"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+)
+
+// getUtxoResult is a simple pair type holding a spend report and error.
+type getUtxoResult struct {
+	report *SpendReport
+	err    error
+}
+
+// GetUtxoRequest is a request to scan for OutPoint from the height BirthHeight.
+type GetUtxoRequest struct {
+	// OutPoint is the target outpoint to watch for spentness.
+	OutPoint *wire.OutPoint
+
+	// BirthHeight is the height at which we expect to find the original
+	// unspent outpoint. This is also the height used when starting the
+	// search for spends.
+	BirthHeight uint32
+
+	// resultChan either the spend report or error for this request.
+	resultChan chan *getUtxoResult
+
+	// result caches the first spend report or error returned for this
+	// request.
+	result *getUtxoResult
+
+	// mu ensures the first response delivered via resultChan is in fact
+	// what gets cached in result.
+	mu sync.Mutex
+
+	quit chan struct{}
+}
+
+// deliver tries to deliver the report or error to any subscribers. If
+// resultChan cannot accept a new update, this method will not block.
+func (r *GetUtxoRequest) deliver(report *SpendReport, err error) {
+	select {
+	case r.resultChan <- &getUtxoResult{report, err}:
```

Comment:
From my reading above, loos like it's possible  for `deliver` to be called twice? As in when/if we detect a spend, but then at the end (I think?) when we call `notifyAllUnspent`.

Reply:
Ah nvm, see the default below :p